OPHD/UI/Core/RadioButtonGroup.h

```diff
@@ -0,0 +1,67 @@
+#pragma once
+
+#include "TextControl.h"
+#include "Label.h"
+#include "../../Cache.h"
+#include "../../Constants.h"
+
+#include <NAS2D/Signal/Signal.h>
+#include <NAS2D/Signal/Delegate.h>
+#include <NAS2D/EventHandler.h>
+#include <NAS2D/Resource/Image.h>
+#include <NAS2D/Utility.h>
+#include <NAS2D/Renderer/Renderer.h>
+#include <NAS2D/MathUtils.h>
+
+#include <algorithm>
+#include <string>
+
+class RadioButtonGroup : public Control
+{
+public:
+	struct ButtonInfo
+	{
+		std::string name;
+		NAS2D::DelegateX<void> delegate;
+	};
+
+	RadioButtonGroup() = default;
+	RadioButtonGroup(std::vector<ButtonInfo> buttonInfos);
+	void setSelected(std::size_t index);
+	void onMove(NAS2D::Vector<int> displacement) override;
+	void update() override;
+	void clearSelection();
+	bool hasOneSelectedItem();
+
+private:
+    class RadioButton : public TextControl
+    {
+    public:
+        using ClickCallback = NAS2D::Signal<>;
+        RadioButton(std::string newText, RadioButtonGroup* parentContainer, NAS2D::DelegateX<void> delegate);
+        ~RadioButton() override;
+        void checked(bool toggle);
+        bool checked() const;
+        void text(const std::string& text);
+        const std::string& text() const;
+        ClickCallback& click();
+        void update() override;
+
+    protected:
+        void onResize() override;
+        void onTextChange() override;
+        void parentContainer(RadioButtonGroup* parent);
+
+    private:
+        const NAS2D::Font& mFont;
+        const NAS2D::Image& mSkin;
+        Label mLabel;
+        ClickCallback mCallback; /**< Object to notify when the Button is activated. */
+        RadioButtonGroup* mParentContainer{nullptr};
+        bool mChecked{false};
+        void onMouseDown(NAS2D::EventHandler::MouseButton button, int x, int y);
```

Comment:
This method should be declared along side the other event handler methods, rather than mixed in with the variables.

This method is probably going to handle the `onClick` event. Probably doesn't matter if it's one event handler, or split into two. Might provide some potentially subclassing benefits if split into two.

Reply:
Moved it into the protected section where the other event handler methods.